src/pages/resultsView/mutation/AddColumns.tsx

```diff
@@ -0,0 +1,263 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import MobxPromise from 'mobxpromise';
+import { observer } from 'mobx-react';
+import { action, computed, makeObservable, observable } from 'mobx';
+import { Checkbox } from 'react-bootstrap';
+import { remoteData, getTextWidth } from 'cbioportal-frontend-commons';
+import { ClinicalAttribute } from 'cbioportal-ts-api-client';
+import { IColumnVisibilityControlsProps } from 'shared/components/columnVisibilityControls/ColumnVisibilityControls';
+import { MSKTab, MSKTabs } from 'shared/components/MSKTabs/MSKTabs';
+import CustomDropdown from 'shared/components/oncoprint/controls/CustomDropdown';
+import AddChartByType from '../../studyView/addChartButton/addChartByType/AddChartByType';
+import { ChartDataCountSet } from '../../studyView/StudyViewUtils';
```

Comment:
When attempting to move `AddChartByType` to shared components, it quickly became apparent that this would be a recursive nightmare, since it itself imports things from `studyView`, which in turn import other things from `studyView` and so on. I think it should be doable, but not without major refactoring. Given that `oncoprint` in `resultsView` already imports the same things from `studyView`, I agree with postponing this to a later PR.